go-sdk/internal/sparse-merkle-tree/node/node.go

```diff
@@ -51,17 +52,27 @@ func NewEmptyNode() core.Node {
 	return &node{nodeType: core.NodeTypeEmpty}
 }
 
-func NewLeafNode(v core.Indexable) (core.Node, error) {
-	n := &node{nodeType: core.NodeTypeLeaf, value: v}
-	// the leaf node's index is calculated as follows:
-	// 1. calculate the index (aka hash) of the value object, call it hV
-	// 2. calculate hash(hV, hV, 1)
-	idx, err := n.value.CalculateIndex()
+func NewLeafNode(s core.Indexable, v ...*big.Int) (core.Node, error) {
```

Comment:
Some comments /check here will be helpful so the consumer of the code knows the function only expects a single value as optional input. Extra values will just be thrown away.

Also, if the code is unlikely to support multiple values, it makes more sense to encourage consumers to pass in `nil` explicitly other than hiding it.